### ISSUE TITLE
Migrating in SQLITE: dropColumn throws error

### ISSUE THREAD

**Issue by wsjwa** [knex/knex#3473]
sqllite Version:
"sqlite3": "4.0.2",

If you have a migration with dropColumn, you'll get the error:
`TypeError: Cannot read property '1' of null`

Because in knex\lib\dialects\sqlite3\schema\ddl.js:
_doReplace() is being called and the Regex searches for "CREATE TABLE..."

Migration:
`    builder.schema.alterTable('USER', table => {
      table.dropColumn('DESCRIPTION');
    });`

Is there a workaround or alternative to dropping columns/changing the column type?

dropColumns() throws the same error btw

**Comment by wsjwa** [comment]
Knex Version is: "0.17.3"

**Comment by lorefnon** [comment]
Can you please share the result of the following (from sqlite shell): 

```sql
select name, sql from sqlite_master where type = "table" and name = "USER"
```

**Comment by wsjwa** [comment]
Result for XC_USER:
`CREATE TABLE XC_USER(ID TEXT NOT NULL PRIMARY KEY, XC_MANDANTEN_ID TEXT NOT NULL, LANDINGPAGE TEXT NOT NULL, NAME TEXT NOT NULL, DESCRIPTION TEXT NOT NULL, EMAIL TEXT NOT NULL, PASSWORD TEXT NOT NULL, CREATED_AT TEXT NOT NULL, CREATED_BY TEXT NOT NULL, LAST_LOGIN TEXT NOT NULL, LANGUAGE TEXT NOT NULL, ACTIVE INTEGER NOT NULL, ADMINISTRATOR INTEGER NOT NULL, GROUP_ADMIN INTEGER NOT NULL, "LANDINGPAGES" integer not null default '0')`

**Comment by wsjwa** [comment]
Should the method even be called at all? Because i'm trying to drop a column and not create a table

**Comment by ricardograca** [comment]
SQLite doesn't support renaming or dropping tables with a single query: https://www.sqlite.org/lang_altertable.html

**Comment by wsjwa** [comment]
> SQLite doesn't support renaming or dropping columns with a single query: https://www.sqlite.org/lang_altertable.html

So Knex is not able to drop Columns in sqlite? Or is it fixed in #3476?

**Comment by kibertoad** [comment]
Released in 0.20.0
